jax/interpreters/ad.py

```diff
@@ -471,6 +480,11 @@ def call_transpose(primitive, params, call_jaxpr, args, ct, _):
   fun = lu.hashable_partial(lu.wrap_init(backward_pass), call_jaxpr)
   fun, out_tree = flatten_fun_nokwargs(fun, in_tree_def)
   params = dict(params, name=wrap_name(params['name'], 'transpose'))
+  if 'donated_invars' in params:
+    new_donated_invars = (*[d for d, x in zip(params['donated_invars'], args)
+                            if not is_undefined_primal(x)],
+                          *[False for x in ct if x is not zero])
```

Comment:
I have a vague sense that we might just be able to make this `True` (assuming that `backward_pass` is only called inside the AD infra) because it only has an effect in the op-by-op grad-of-jit case, and in that case using donation for cotangent buffer accumulation is always a good idea. But let's follow up and not make that change now!

Reply:
Hey nice catch, I think you're right!